### PR TITLE
Add `download` and `upload` subcommands to `airgap_script`

### LINKED ISSUE #228: Add Cyberwatch Air-gapped assets with a one liner command
Add two binaries to have a one liner command to fetch Cyberwatch script and one to upload script result.

### PR DESCRIPTION
**Pull request by polyedre**
This PR adds the `airgap_script` resource to the command line interface. It adds `download` and `upload` subcommand.

The workflow is:

1. Download the scripts (the default destination is `Scripts`).

```sh
cyberwatch-cli airgap download-scripts --dest-dir /tmp/scripts
# Or this (default destination is `scripts`)
cyberwatch-cli airgap download-scripts
```

2. Run the scripts.

```sh
mkdir assets
/tmp/script/Linux/run > "assets/$(hostname)"
```

3. Upload the results.

```sh
cyberwatch-cli airgap upload assets/*
```

Closes #228

### CODE REVIEW

**Comment by hezanathos** [comment]
```cyberwatch-cli airgap upload assets/*```

Can we automatically push all contained files if we specify a folder ?
We have to make the procedure as light as possible.

File: docs/cli.md [hunk 1]
```diff
@@ -126,3 +126,51 @@ cyberwatch-cli docker-image show vulnerabilities --format junit-xml
...
+```
+
+To specify a different destination directory:
+
+```sh
+export SCRIPTS_DIR=/tmp/scripts
+cyberwatch-cli airgap download-scripts --dest-dir $SCRIPTS_DIR
+```
+
+Please note that the destination must be non-existent or the command will fail.
```
[the remark this answers is not in the document]
Comment (polyedre) [reply docs/cli.md:150]: This was requested by @fmang
What is better behaviour for you? That the script override the folder? What to do if a script was deleted in the Cyberwatch instance and is present in the `scripts` folder before the download?

**Comment by polyedre** [comment]
> `cyberwatch-cli airgap upload assets/*`
> 
> Can we automatically push all contained files if we specify a folder ?
> We have to make the procedure as light as possible.

I can add this. So the behavior:

If `target` is a folder, upload all files in it. If `target` is a list of files, upload them.

**Comment by polyedre** [comment]
A discussed over the phone with @hezanathos, here is the behavior chosen:

### Download-scripts

```sh
cyberwatch-cli airgap download-scripts [--dest-dir DESTINATION]
```
By default, create a folder named `cyberwatch-airgap` with the following tree structure. The `scripts` folder contains downloaded scripts.

```
cyberwatch-airgap
├── scripts
│   ├── aix
│   │   ├── InfoScript.sh
│   │   ├── PortScript.sh
│   │   └── run
│   ├── linux
│   │   ├── InfoScript.sh
│   │   ├── PortScript.sh
│   │   └── run
│   └── windows
└── uploads
```

Each `run` script executes **only** the script downloaded in the last version. This ensures old scripts are not executed if still present in this directory. If the destination directory already exists, the content is overwritten. There is no prompt because the default destination is unique enough.

### Upload

```
cyberwatch-cli airgap upload [FILE1 FILE2 FILE3...]
```
If no argument is provided to the `upload` subcommand, the script default to upload all the results of scripts in `cyberwatch-airgap/uploads/*` if the directory exists. If some files are specified, only these files are uploaded.

File: cbw_api_toolbox/cbw_api.py [hunk 2]
```diff
@@ -66,11 +66,12 @@ def _cbw_parser(self, response):
         """Parse the response text of an API request"""
         try:
             result = json.loads(response.text, object_hook=lambda d: namedtuple('cbw_object', d.keys())(*d.values()))
+            return result
         except TypeError:
             self.logger.error("An error occurred while parsing response")
-        return result
+            sys.exit(-1)
```
Comment (polyedre) [inline cbw_api_toolbox/cbw_api.py:72]: I'm not sure about this modification. The motivation was to avoid having `None` as a return value for this. But this is not related at all to this PR and exiting here might not be the best behavior.

**Comment by fenrisl** [comment]
I would add a small section for powershell script execution like :

---
To execute the scripts on a windows machine:

```powershell

.\cyberwatch-airgap\scripts\Windows\run.ps1 > .\cyberwatch-airgap\uploads\${env:COMPUTERNAME}
```
----
Also while testing on Windows 10, after copying the result and trying to upload I had this error : 
```sh
'utf-8' codec can't decode byte 0xff in position 0: invalid start byte
```

**Comment by polyedre** [comment]
The encoding problem was that on PowerShell 5, the default encoding is UTF-16. I edited the PR to add documentation and the use of `chardet` to detect the encoding.
